sigstore/_internal/rekor/client.py

```diff
@@ -199,8 +200,26 @@ def __init__(self, url: str, pubkey: bytes, ctfe_pubkey: bytes) -> None:
             {"Content-Type": "application/json", "Accept": "application/json"}
         )
 
-        self._pubkey = serialization.load_pem_public_key(pubkey)
-        self._ctfe_pubkey = serialization.load_pem_public_key(ctfe_pubkey)
+        pubkey = serialization.load_pem_public_key(pubkey)
+        if not isinstance(
+            pubkey,
+            (
+                ec.EllipticCurvePublicKey,
+            ),
+        ):
+            raise RekorClientError(f"Invalid public key type: {pubkey}")
+        self._pubkey = pubkey
```

Comment:
Flagging for review: this was already an implicit constraint (due to how we do the SET verification; it assumes an EC key in the cryptography APIs we use), but we didn't catch it until we switched to the upstream cryptography type hints.

I think this is _technically_ overly conservative of us, as the Rekor root pubkey could be RSA instead. But neither the production nor staging instance currently does that (and both of the keys we have checked in are EC).

Reply:
There has actually been some discussion about changing the key to RSA if I understand correctly, cc @haydentherapper @asraa.

Reply:
Hmmmm I don't think so. Maybe you are confusing it with me debating (no longer) this week on changing our TUF root management sign-off keys to RSA for better compatibility. But we are good to go on that front as well.

On my end you're fine here. The cosign code presumes Rekor to always be ecdsa too

Reply:
Oh, I was mixing those up. Thanks!

Reply:
Right now, there are some assumptions in Rekor and Cosign that mandate that the key be EC. This is not true with CT however, which is why you've seen RSA and EC keys for staging/prod respectively. 

It'd be nice to support different types of keys in Rekor if possible, but this is not planned to be implemented any time soon. (I also prefer EC far more due to the reduced signature size).

Reply:
Thanks @asraa and @haydentherapper!